redins.py

```diff
@@ -1,244 +1,185 @@
-    
-    graph=facebook.GraphAPI(access_token=page_access_token,version="3.0")
-
-    dirs = __os.listdir(__os.path.join(curr_dir, 'red_media'))
-
-    print("facebook upload")
-    for j in range(num):
-        try:
-            files = __random.choice(dirs)
-            files = __os.path.join(curr_dir, 'red_media', files)
-            photo=open(files,"rb")
-            print("posting",files)
-
-            graph.put_photo(
+            photo = open(file, "rb")
+            self.facebook_client.put_photo(
                 image=photo,
-                message=__random.choice(cap),
+                message=random.choice(self.caption.facebook_caps),
             )
             photo.close()
+            print(f"Uploaded {file} to Facebook..")
         except Exception as e:
-                print("Error occured: {}" .format(str(e)))
-
-def start_upload():
-    print('Starting upload')
-    num_of_uload = int(input("Enter the number of files to be uploaded: "))
-    if config.IG:
-        print('Uploading to IG')
-        uload_to_ig(num_of_uload)
-    if config.TWITTER:
-        print('Uploading to Twitter')
-        uload_to_twitter(num_of_uload)
-    if config.FB:
-        print('Uploading to FB')
-        uload_to_fb(num_of_uload)
-    else:
-        print('No upload')
-
-def main():
-    if check_folder():
-        config.get_data()
-        if len(__os.listdir('red_media'))>2:
-            start_upload()
-            return
-        
-        for j in JSONs:
-            get_links(j)
-            write_meme()
-            dload(j)
-            start_upload()
-    else:
-        print("Error has occured in creating file")
-
-    
+            print("Error occured: {}" .format(str(e)))
+
+    def start_upload(self, file):
+        functions = {"instagram": self.upload_to_ig,
+                     "twitter": self.upload_to_twitter,
+                     "facebook": self.upload_to_fb}
+
+        for module, function in functions.items():
+            if getattr(self, module):
+                print(f"Uploading {file} to ", module)
+                function(file)
+
+    def main(self):
+        self.redpy.createFolder()
+        files_to_upload = glob.glob('*.jpg')
+        local_upload = True
+        num_of_upload = int(
+            input("Enter the number of files to be uploaded: "))
+        if len(files_to_upload) == 0:
+            for sub in self.config.subreddits:
+                files_to_upload = self.redpy._getImages(
+                    self.redpy._generateJSON(
+                        sub, self.config.sub_category), num_of_upload)
+            local_upload = False
+        random.shuffle(files_to_upload)
+        max_limit = len(files_to_upload)
+        if num_of_upload > max_limit:
+            print("Files insufficient")
+            print("Setting upload limit to ", max_limit)
+            num_of_upload = max_limit
+        files_to_upload = files_to_upload[:num_of_upload+1]

```

Comment:
> files_to_upload = files_to_upload[:num_of_upload+1]

Isn't this a bug? `list[:n]` will return `n` items list. Your code will return n+1 files